include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -31,13 +32,104 @@ __is_power_of_two(std::size_t __number)
     return (__number != 0) && ((__number & __number - 1) == 0);
 }
 
+// can't use std::shared_ptr, because commonly sizeof(std::shared_ptr) is 2*sizeof(void*) while
+// sizeof(__block_header) must be power of 2 and with std::shared_ptr instead of
+// __sycl_device_shared_ptr memory fragmentation increases drastically
+class __sycl_device_shared_ptr
+{
+    // state is shared between TU with the context and all objects allocated in this TU
+    struct __shared_device
+    {
+        std::optional<sycl::device> _M_device;
```

Comment:
Can we simplify those nested optional-s?

Reply:
It seems `optional` are not needed, as call to `__get_policy()`/`__get_device_ptr()` after dtor and before ctor calls is equivalent to impossibility to get offload device with required type.